### ISSUE TITLE
Reorganize Races

### ISSUE THREAD

**Issue by Vinkdicator** [Return-To-The-Roots/s25client#1147]
I think it might be handy to have the race textures stored in individual files/folders in a 'race' folder similar to how game maps are stored. This would enable players to add their own races. If someone uses a race that is unknown to the other player on multiplayer, it could simply default to one of the default races.

**Comment by Spikeone** [comment]
For that, loading graphics needs to be cleaned up - I think @Flamefire already put a lot of effort into that but I don't know if that'd be possible already or if it's just a lot of work.

**Comment by Flamefire** [comment]
Yep I wanted to get something like that but it's not that easy: We can't provide folders with the original races due to copyright. They are in multiple packed files in S2 format. Furthermore it is not a 1:1 replacement of graphics: Different races use different animations (length and images and number of frames). What I worked on was having a description file in LUA format for races, buildings, workers... but it is a real lot of work and gets very complicated, e.g. you got winter or summer textures. And there are hundreds of images per race. It will also be a performance hit when data is read from dynamic locations rather than hardcoded as it is currently done. So this has to be kept in mind when designing the system

TLDR: I have a draft but nothing finished yet and the draft is not good enough yet either.
